src/query/functions/temporal/base.go

```diff
@@ -81,12 +81,23 @@ func (o baseOp) Node(controller *transform.Controller, opts transform.Options) t
 // baseNode is an execution node
 type baseNode struct {
 	op            baseOp
-	controller    *transform.Controller
+	controller    controller
 	cache         *blockCache
 	processor     Processor
 	transformOpts transform.Options
 }
 
+// controller is a minimal interface of transform.Controller, used exclusively
```

Comment:
Not a huge fan of having test specific stuff in our code. Anyway we can do this without putting this here?

Reply:
Agreed--let me explain what I'm trying to do; maybe there's a good way to do it without this :)

I want to test that blocks we create are closed (especially in case of an error). That means that I need a block that allows me to make that assertion; regular blocks don't (which is reasonable). Since we create blocks via `controller.NewBlockBuilder`, `controller` is my entry point into injecting a new block. Therefore, I mock out controller.

Alternatives considered: 
- create a public transform.Controller interface, and use that everywhere instead. I didn't think it was worth refactoring every usage of controller for this, but we could.

Reply:
So it's not too bad, but given that this is a bug fix for a panic, I'd rather get this in first and do the refactor after. Thoughts?

Reply:
If we're going more hotfix then I'd say jam it with no test; if it's not a hotfix wouldn't mind having the refactor in, but will be happy enough with pulling this into an isolated file then filing an explicit issue if that sounds good?

Reply:
Sure thing--filed https://github.com/m3db/m3/issues/1430 to track. Will do that shortly!